src/cmd/tools/zarf.go

```diff
@@ -155,31 +155,17 @@ var updateCredsCmd = &cobra.Command{
 
 			// Update artifact token (if internal)
 			if slices.Contains(args, message.ArtifactKey) && newState.ArtifactServer.PushToken == "" && newState.ArtifactServer.IsInternal() {
-				tunnel, err := c.NewTunnel(cluster.ZarfNamespaceName, cluster.SvcResource, cluster.ZarfGitServerName, "", 0, cluster.ZarfGitServerPort)
-				if err != nil {
-					return err
-				}
-				_, err = tunnel.Connect(cmd.Context())
-				if err != nil {
-					return err
+				pushToken, err := c.UpdateInternalArtifactServerToken(ctx, oldState.GitServer)
+				updateErr := fmt.Errorf("unable to create the new Gitea artifact token: %w", err)
+				// TODO once Zarf is changed so the default state is empty for a service when it is not deployed
+				// and sufficient time has passed for users state to get updated we should error on isNotFound
+				if err != nil && !kerrors.IsNotFound(err) {
+					return updateErr
 				}
-				defer tunnel.Close()
-				tunnelURL := tunnel.HTTPEndpoint()
-				giteaClient, err := gitea.NewClient(tunnelURL, oldState.GitServer.PushUsername, oldState.GitServer.PushPassword)
-				if err != nil {
-					return err
-				}
-				err = tunnel.Wrap(func() error {
-					tokenSha1, err := giteaClient.CreatePackageRegistryToken(ctx)
-					if err != nil {
-						return err
-					}
-					newState.ArtifactServer.PushToken = tokenSha1
-					return nil
-				})
-				if err != nil {
-					// Warn if we couldn't actually update the git server (it might not be installed and we should try to continue)
-					message.Warnf(lang.CmdToolsUpdateCredsUnableCreateToken, err.Error())
+				if kerrors.IsNotFound(err) {
+					message.Warn(updateErr.Error())
```

Comment:
can we return errs as errs plz :)

Reply:
The point of this PR is to log a warning instead of an error when `kerrors.IsNotFound` is true since otherwise `zarf tools update-creds` fails every time.

An alternative I considered was to create a function `internalGitServerExists()` to use here instead of `isInternal()` which would avoid this block all together if it's not found. I ultimately went with this way, as I thought two functions might be confusing (though maybe that's okay since the reality is that it's confusing and we need to change how we create state)

Reply:
okay, that makes sense. :)